src/screenComponents/shieldsEnableButton.cpp

```diff
@@ -31,20 +40,46 @@ void GuiShieldsEnableButton::onDraw(sf::RenderTarget& window)
     {
         if (my_spaceship->shield_calibration_delay > 0.0)
         {
+            // If the shields are calibrating, show a progress bar.
             button->hide();
             bar->show();
             bar->setValue(my_spaceship->shield_calibration_delay);
         }
         else
         {
+            // Show the button, hide the bar, set button state to shield activation state.
             button->show();
             button->setValue(my_spaceship->shields_active);
             bar->hide();
-            string shield_status=my_spaceship->shields_active ? tr("shields","ON") : tr("shields","OFF");
+
+            // Format strings based on settings and shield activation state.
+            string shield_frequency = icon_only ? frequencyToString(my_spaceship->shield_frequency) : tr("{frequency} Shields: ").format({{"frequency", frequencyToString(my_spaceship->shield_frequency)}});
+            string shield_status = my_spaceship->shields_active ? tr("shields", "ON") : tr("shields", "OFF");
+
+            // If the game's using beam frequencies, show the shield's current frequency setting.
+            // If the button's only showing the icon, don't bother with long, formatted strings.
             if (gameGlobalInfo->use_beam_shield_frequencies)
-                button->setText(tr("{frequency} Shields: {status}").format({{"frequency", frequencyToString(my_spaceship->shield_frequency)}, {"status", shield_status}}));
+            {
+                if (icon_only)
+                {
+                    button->setText(shield_frequency);
+                }
+                else
+                {
+                    button->setText(string("{frequency} {status}").format({{"frequency", shield_frequency}, {"status", shield_status}}));
```

Comment:
This seems to have changed from `"{frequency} Shields: {status}"` to `"{frequency} {status}"`?

Reply:
Yes. The logic of whether to add "Shields: " to the string is in L56.

Reply:
Ah, I missed that. That still looks odd then. In case a translation needs "{frequency} {status} shields" that is now no longer possible.

Reply:
I don't understand the problem, so I'll just make it more verbose so it changes less